src/pages/dataElements/form/fields.tsx

```diff
@@ -328,33 +388,127 @@ export function LegendSetField() {
     )
 }
 
+/**
+ *
+ * ValueType
+ * * Field rule: Unless valueType or the selected optionSet's valueType is
+ *   MULTI_TEXT, filter out the MULTI_TEXT option
+ * * Field rule: When the selected optionSet's valueType is MULTI_TEXT, disable
+ *   valueType field
+ * * Field rule: When the selected optionSet's valueType is MULTI_TEXT, set
+ *   valueTypeField value to optionSet's valueType
+ *
+ */
+const valueTypeHelpText = i18n.t('The type of data that will be recorded.')
+const valueTypeDisabledHelpText = i18n.t(
+    'Disabled as the value type must match the value type of the selected option set'
+)
+const valueTypeOptionSetFields = ['id', 'valueType']
 export function ValueTypeField() {
+    const { change } = useForm()
+    const { values } = useFormState({ subscription: { values: true } })
+    const disabled = !!values.optionSet.id
+    const [lazy] = useState(!values.optionSet.id)
+    const { refetch, ...optionSetQuery } = useOptionSetQuery({
+        lazy,
+        variables: {
+            id: values.optionSet.id,
+            fields: valueTypeOptionSetFields,
+        },
+    })
+
+    useEffect(() => {
+        if (values.optionSet.id) {
+            refetch({ id: values.optionSet.id })
+        }
+    }, [refetch, values.optionSet.id])
```

Comment:
I'm a bit worried about the need of doing this, but it does make sense - especially since `valueType` is not present in the optionSet request for now - but ideally we could just reuse that and set cacheTime to not send new requests in this field. I'm just worried about the requests that would go out if a lot of fields would need to do something like this. I'm also not a fan of the fetch logic in this field - but that is something we could put in a hook - but not sure if it's worth it here?

Reply:
>  but ideally we could just reuse that and set cacheTime to not send new requests in this field

That's what @ismay suggested as well. The problem is that the option sets request from the `OptionSetSelect` is paginated, so we'd have to know which page that particular option set is in and simply requesting what we need here felt a lot simpler.

Another point for doing a request here is that we colocate the data dependency with where it's actually used/needed. This way we don't put a field that we don't need in the option set field into the requests for its options (which might actually result in higher bandwidth usage than the request proposed by this PR).

> I'm also not a fan of the fetch logic in this field - but that is something we could put in a hook - but not sure if it's worth it here?

Yet another case for just using react-query because it supports `enabled` rather than `lazy`, which would remove the fetch logic entirely. But I didn't use it because we're not allowed :upside_down_face: 

> I'm just worried about the requests that would go out if a lot of fields would need to do something like this.

I'm not worried about this until we have to be. These requests should transfer only a couple of bytes and are used by administrators only so that - even when the entire dhis2 instance is used with a limited amount of data - this should be next to insignificant. I def. agree that once we have heavy requests we should think about whether it makes sense to do it the same way as here, but I wouldn't optimize something that might not need optimization after all.

Reply:
> That's what @ismay suggested as well. The problem is that the option sets request from the OptionSetSelect is paginated, so we'd have to know which page that particular option set is in and simply requesting what we need here felt a lot simpler.

I see, that makes sense. 

> Yet another case for just using react-query because it supports enabled rather than lazy, which would remove the fetch logic entirely. 

Agreed, to be pedantic though - it's due to refetch and static queries - lazy is just a "rename" of enabled. But yeah, it's due to how we have to imperatively call refetch. 

> Another point for doing a request here is that we colocate the data dependency with where it's actually used/needed.

Here it's just one. But some other fields (in other sections) will have a lot of dependencies, to the point that I think it would make sense to fetch all of those in one request - even could make sense to do it upfront. But it could potentially look the same - and extract it to a hook that is used across multiple fields. But lets keep this for now.